### PR TITLE
add dynamic fee template

### PR DESCRIPTION
**Pull request by ratik21**

### CODE REVIEW

File: examples/ref-templates/assets/dynamic-fee.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    - Suppose the owner of account A wants to send a payment to account TMPL_TO, but does not want 
+      to pay a transaction fee. If account A signs the following contract with the appropriate parameters 
+      (specifying all of the necessary details of the payment transaction), then anyone can cover a fee 
+      for that payment on account A's behalf.
+
+    - The first transaction must spend the transaction fee into account A, and the second transaction 
+      must be the specified payment transaction from account A to account TMPL_TO.  
+
+
```
Comment (amityadav0) [inline examples/ref-templates/assets/dynamic-fee.py:28]:
```suggestion
```

File: examples/ref-templates/assets/dynamic-fee.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    # Next, specify that the receiver of funds from the first transaction is equal to the sender 
+    # of the second transaction (since the first transaction is paying the second transaction's fee)
+    address_check = Gtxn[0].receiver() == Gtxn[1].sender()
+
+    # Next, check that the first transaction's amount is equal to the fee of the second transaction.
+    amount_check = Gtxn[0].amount() == Gtxn[1].fee()
+
+    #Check that the second transaction is a payment.
+    secondTxn_pay_check = Gtxn[1].type_enum() == TxnType.Payment
+
```
Comment (amityadav0) [inline examples/ref-templates/assets/dynamic-fee.py:55]:
```suggestion
```

File: examples/ref-templates/assets/dynamic-fee.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    # fold all the above checks into a single boolean.
+    required_condition = And(
+        grp_check,
+        firstTxn_pay_check,
+        address_check,
+        amount_check,
+        secondTxn_pay_check
+    )
+
```
Comment (amityadav0) [inline examples/ref-templates/assets/dynamic-fee.py:65]:
```suggestion
```

File: examples/ref-templates/assets/dynamic-fee.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    params_condition = And(
+        recv_field_check,
+        cls_field_check,
+        amount_field_check,
+        #fv_field_check,
+        #lv_field_check,
+        #lease_field_check
+    )
+
```
Comment (amityadav0) [inline examples/ref-templates/assets/dynamic-fee.py:94]:
```suggestion
```

File: examples/ref-templates/scripts/dynamic-fee.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+  // setup a contract account and send 1 ALGO from master
+  await deployer.fundLsig("dynamic-fee.py", [], { funder: masterAccount, fundingMicroAlgo: 100000000 }, { closeRemainderTo: masterAccount.addr }); 
+
+  let contract = await deployer.loadLogic("dynamic-fee.py");
+  const escrow = contract.address(); //contract account
+
+  const signedContract =  await deployer.loadLogic("dynamic-fee.py");
+  signedContract.sign(masterAccount.sk); //sign contract 
+
```
Comment (amityadav0) [inline examples/ref-templates/scripts/dynamic-fee.js:36]:
```suggestion
```

File: examples/ref-templates/scripts/dynamic-fee.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+  let transactions = [
+    mkTxnParams(masterAccount, escrow, 1000, signedContract, {}),
+    mkTxnParams({ addr: escrow}, johnAccount.addr, 700000, contract, { totalFee: 1000, closeRemainderTo: bobAccount.addr })]
+
+  //Group Transaction FAIL - Correct transaction Fee is used BUT closeRemainderTo is set to bob
+  await transferMicroAlgoAtomic(deployer, transactions);
+
```
Comment (amityadav0) [inline examples/ref-templates/scripts/dynamic-fee.js:44]:
```suggestion
```

File: examples/ref-templates/assets/dynamic-fee.py [hunk 7]
```diff
@@ -0,0 +1,100 @@
+from pyteal import *
+
+#replace the following params with your custom values
+john = Addr("2UBZKFR6RCZL7R24ZG327VKPTPJUPFM6WTG7PJG2ZJLU234F5RGXFLTAKA")
+master = Addr("JAKNKAMEW63I3H3AXWINXEWW6CCLKX4MQGNTTNBSSUHNCHSLQURMMO47XM")
```
Comment (amityadav0) [inline examples/ref-templates/assets/dynamic-fee.py:5]: Please change this address to master which is in `algob.config.js`

File: examples/ref-templates/assets/dynamic-fee.py [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    The contract works by approving a group of two transactions (meaning the two transactions will occur together or not at all). 
+
+    - Suppose the owner of account A wants to send a payment to account TMPL_TO, but does not want 
+      to pay a transaction fee. If account A signs the following contract with the appropriate parameters 
+      (specifying all of the necessary details of the payment transaction), then anyone can cover a fee 
+      for that payment on account A's behalf.
+
+    - The first transaction must spend the transaction fee into account A, and the second transaction 
+      must be the specified payment transaction from account A to account TMPL_TO.  
+
```
Comment (amityadav0) [inline examples/ref-templates/assets/dynamic-fee.py:27]:
```suggestion
```

File: examples/ref-templates/assets/dynamic-fee.py (hunk 7 above)
Comment (ratik21) [reply examples/ref-templates/assets/dynamic-fee.py:5]: ok

File: examples/ref-templates/scripts/dynamic-fee.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  // setup a contract account and send 1 ALGO from master
+  await deployer.fundLsig("dynamic-fee.py", [], { funder: masterAccount, fundingMicroAlgo: 100000000 }, { closeRemainderTo: masterAccount.addr }); 
+
+  let contract = await deployer.loadLogic("dynamic-fee.py");
+  const escrow = contract.address(); //contract account
+  
+  await deployer.mkDelegatedLsig("dynamic-fee.py", [], masterAccount); // sign contract
+  const signedContract =  await deployer.getDelegatedLsig('dynamic-fee.py');
+  
+
```
Comment (amityadav0) [inline examples/ref-templates/scripts/dynamic-fee.js:37]:
```suggestion
```

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**